app/src/main/kotlin/github/com/raccok/dota2androidapp/MainActivity.kt

```diff
@@ -22,108 +22,114 @@
                     Toast.LENGTH_LONG).show()
 
-            textView.text = resources.getString(R.string.load_pref_hero_success_message, favoriteHero)
+            textView.text = resources.getString(R.string.favorite_hero_message, favoriteHero)
         } else {
-            // Show a dialog to ask for the user's favorite hero
-            var mMainAlert = AlertDialog.Builder(this)
-            mMainAlert.setTitle(resources.getString(R.string.welcome_dialog_title)).setMessage(resources.getString(R.string.welcome_dialog_message))
-
-            val input = EditText(this)
-            mMainAlert.setView(input)
-
-            mMainAlert.setPositiveButton(resources.getString(R.string.dialog_ok)) { _, _ ->
-
-                viewModel.getHero(input.text.toString())
+            val alert = AlertDialog.Builder(this)
+            alert.setTitle(resources.getString(R.string.welcome_dialog_title))
+            alert.setMessage(resources.getString(R.string.welcome_dialog_message))
+            val inputField = EditText(this)
+            alert.setView(inputField)
+            alert.setPositiveButton(resources.getString(R.string.dialog_ok)) { _, _ ->
+                mViewModel.getHero(inputField.text.toString())
             }.setNegativeButton(resources.getString(R.string.dialog_cancel)) { _, _ -> }
-
-            mMainAlert.show()
+            alert.show()
         }
     }
 
-    private fun validateUserInput(userInput: String, listOfHeros: List<HeroEntity>) {
-        if (!listOfHeros.isEmpty()) {
+    // Check if userInput is a valid (currently available) Dota 2 hero.
+    // If it is, save it to device storage. If it isn't, return to the input dialog.
+    private fun validateUserInput(userInput: String, listOfHeroes: List<HeroEntity>) {
+        if (listOfHeroes.isNotEmpty()) {
             //TODO: move all operations pertaining to preferences to another class
-            // Put it into memory (don't forget to commit!)
-            val e = mAppPrefsGeneral().edit()
+            val e = appPrefsGeneral().edit()
             e?.putString(PreferenceStrings.PREF_FAV_HERO, userInput)
-            e?.commit()
+            e?.apply()
 
             Toast.makeText(this, resources.getString(R.string.success_pref_hero_commit, userInput),
                     Toast.LENGTH_LONG).show()
-            textView.text = resources.getString(R.string.load_pref_hero_success_message, userInput)
+            textView.text = resources.getString(R.string.favorite_hero_message, userInput)
         } else {
             Toast.makeText(this, resources.getString(R.string.error_invalid_hero, userInput),
                     Toast.LENGTH_LONG).show()
+            initialDisplay()
         }
     }
 
+    // Try to fetch currently available Dota 2 heroes from the official Dota 2 API.
     private fun observeLiveData() {
-        viewModel.isLoadingLiveData.observe(this, Observer<Boolean> {
-            it?.let {
-                status ->
-            }
+        // TODO: TEST
```

Comment:
I will resolve all TODOs later and test this when merging https://github.com/Rhacco/Dota2AndroidApp/pull/2